core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/ProjectionRemovealOptimizer.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.query.algebra.evaluation.impl;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.rdf4j.query.BindingSet;
+import org.eclipse.rdf4j.query.Dataset;
+import org.eclipse.rdf4j.query.algebra.Projection;
+import org.eclipse.rdf4j.query.algebra.TupleExpr;
+import org.eclipse.rdf4j.query.algebra.Var;
+import org.eclipse.rdf4j.query.algebra.evaluation.QueryOptimizer;
+import org.eclipse.rdf4j.query.algebra.helpers.AbstractQueryModelVisitor;
+
+/**
+ * If a projection node in the algebra does not contribute or change the results it can be removed from the tree.
+ *
+ * For example <code>
+ * SELECT ?s ?p ?o
+ * WHERE {?s ?p ?o }
+ * </code> Does not need a projection as the inner statement pattern returns the same result.
+ * 
+ * While <code>
+ *  * SELECT ?s ?p 
+ * WHERE {?s ?p ?o }
+ * </code> Does as the statement pattern has one more variable in use than the projection.
+ * 
+ * @author Jerven Bolleman
+ */
+public class ProjectionRemovealOptimizer implements QueryOptimizer {
```

Comment:
Minor typo, should be `ProjectionRemovalOptimizer`

Reply:
Ai. Will fix